pkg/controllers/reallocation/controller.go

```diff
@@ -78,6 +78,11 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	if err := c.utilization.terminateExpired(ctx, provisioner); err != nil {
 		return reconcile.Result{}, fmt.Errorf("marking nodes terminable, %w", err)
 	}
+
+	// 5. Delete any node that has been unable to come up for 15 minutes.
+	if err := c.utilization.terminateFailedToBecomeReady(ctx, provisioner); err != nil {
```

Comment:
This will be skipped if `provisioner.Spec.TTLSecondsAfterEmpty` is not specified.  I think we'd want to do this no matter what?

Reply:
You're right. Although I don't know if we want to make a whole new controller for this. I could make it so that this function runs regardless of the specification of TTLSecondsAfterEmpty, but that might confuse those reading the code base. WDYT?

Reply:
I think we need to step back and try to come up with the core logical concepts behind this, and how the behaviors relate to each other.